module/VuFind/src/VuFind/Db/Service/SearchService.php

```diff
@@ -265,6 +295,20 @@ public function getSavedSearchesWithMissingChecksums(): array
      */
     public function deleteExpired(DateTime $dateLimit, ?int $limit = null): int
     {
-        return $this->getDbTable('search')->deleteExpired($dateLimit->format('Y-m-d H:i:s'), $limit);
+        $subQueryBuilder = $this->entityManager->createQueryBuilder();
+        $subQueryBuilder->select('s.id')
+            ->from($this->getEntityClass(SearchEntityInterface::class), 's')
+            ->where('s.created < :dateLimit')
+            ->setParameter('dateLimit', $dateLimit->format('Y-m-d H:i:s'));
```

Comment:
Do we need to format the date limit, or can we pass the DateTime object directly?